extensions/amazon-aws/src/console.tsx

```diff
@@ -17,7 +22,15 @@ export default function Console() {
           icon={service.icon.path}
           actions={
             <ActionPanel>
-              <AwsAction.Console url={`${AWS_URL_BASE}${service.arg}`} />
+              <AwsAction.Console
+                url={
+                  typeof process.env.AWS_SSO_ACCOUNT_ID !== "undefined" &&
+                  typeof process.env.AWS_SSO_ROLE_NAME !== "undefined" &&
+                  typeof process.env.AWS_SSO_START_URL !== "undefined"
+                    ? `${process.env.AWS_SSO_START_URL}console/?account_id=${process.env.AWS_SSO_ACCOUNT_ID}&role_name=${process.env.AWS_SSO_ROLE_NAME}&destination=${AWS_URL_BASE}${service.arg}`
```

Comment:
According to the [doc](https://docs.aws.amazon.com/singlesignon/latest/userguide/createshortcutlink.html?icmpid=docs_sso_console#constructing-shortcut-links):

> All parameter values of the URL, including the account ID, permission set name, and destination URL, must be URL-encoded.

So we should be wrapping all params within `encodeURI()`.

Reply:
I don't think the entire URL needs to be encoded, just the parameter, so basically:

```typescript
`${process.env.AWS_SSO_START_URL}console/?account_id=${encodeURI(process.env.AWS_SSO_ACCOUNT_ID)}&role_name=${encodeURI(process.env.AWS_SSO_ROLE_NAME)}&destination=${encodeURI(AWS_URL_BASE + service.arg)}`
```

Reply:
Actually you will also need to normalize the URL, so we need a function like:

```typescript
function normalizeUrl(url: string): string {
  if (url.endsWith('/')) {
    return url;
  }
  return `${url}/`;
}
```

and then use:

```typescript
`${normalizeUrl(process.env.AWS_SSO_START_URL)}console?account_id=${encodeURI(process.env.AWS_SSO_ACCOUNT_ID)}&role_name=${encodeURI(process.env.AWS_SSO_ROLE_NAME)}&destination=${encodeURI(AWS_URL_BASE + service.arg)}`
```

Reply:
You can resolve this in favor of the new comment.